### DISCUSSION TITLE
How to work with Azure App Secrets

### DISCUSSION THREAD

**Discussion by nickna** [microsoft/aspire#3289] [General]
Hi Folks,

I'm trying to reference an Azure Key Vaults with the Aspire Security.KeyVault package (Preview 4). 
`AddAzureKeyVaultSecrets()` is missing in package `Aspire.Azure.Security.KeyVault`, though it's mentioned in the [docs](https://learn.microsoft.com/en-us/dotnet/aspire/security/azure-security-key-vault-component?tabs=dotnet-cli). 
Also, `Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs` throws an exception because it's not finding vaultUrl in appsettings.json. 

Is this a known issue, or am I making a rookie mistake here? :-) 

-----

Sample **AppHost\Program.cs**
`var secrets = builder.AddAzureKeyVault("secrets");`

Sample **AppHost\appSettings.json**
```
{
    "Aspire": {
        "Azure": {
            "Security": {
                "KeyVault": {
                    "VaultUri": "https://VaultNameHere.vault.azure.net/",
            "HealthChecks": true,
            "Tracing": false,
            "ClientOptions": {
                        "DisableChallengeResourceVerification": true
            }
                }
            }
        }
    }
}
```

**Comment by davidfowl** [comment]
In development if you are using an existing keyvault then you can type:

```C#
var secrets =
    builder.ExecutionContext.IsPublishMode
    ? builder.AddAzureKeyVault("secrets")
    : builder.AddConnectionString("secrets");
```

This says, using an existing keyvault when running, but then provision a keyvault when we publish.
